decompiler/frontend/binaryninja/handlers/constants.py

```diff
@@ -29,72 +14,71 @@ def register(self):
-        ):
-            return self._lift_symbol_pointer(bv, address, symbol)
-
-        if (
-            not isinstance(pointer, mediumlevelil.MediumLevelILImport)
-            and (symbol is None or symbol.type != SymbolType.DataSymbol)
-            and (string := bv.get_string_at(address, partial=True) or bv.get_ascii_string_at(address, min_length=2))
-        ):
-            return Constant(address, Pointer(Integer.char()), Constant(string.value, Integer.char()))
-
-        if (variable := bv.get_data_var_at(address)) is not None:
-            return self._lifter.lift(variable, bv=bv, parent_addr=None)
-
-    def _lift_symbol_pointer(self, bv: BinaryView, address: int, symbol: bSymbol) -> Optional[Union[Symbol, UnaryOperation]]:
-        """Try to lift a pointer at the given address with a Symbol as a symbol pointer."""
-        # symbol containing custom function name, e.g. lookup
-        if symbol.type == SymbolType.FunctionSymbol:
-            return FunctionSymbol(symbol.name, address, vartype=Pointer(Integer.char()))
-        # symbol containing c function name, Win API name (tailcall), etc. e.g. memset
-        if symbol.type in (SymbolType.ImportedFunctionSymbol, SymbolType.ExternalSymbol):
-            return ImportedFunctionSymbol(symbol.name, address, vartype=Pointer(Integer.char()))
-        # IAT entry or .got entry
-        if symbol.type == SymbolType.ImportAddressSymbol:
-            # if it is .got entry, then it points at the external global variable or function
-            # so we try to get the symbol on that address, lift it and lift .got entry as an address of that (lifted) symbol
-            pointer_value = bv.read_pointer(symbol.address)
-            global_data_pointed_by_symbol = bv.get_data_var_at(pointer_value)
-            if global_data_pointed_by_symbol:
-                lifted_global_var = self._lifter.lift(global_data_pointed_by_symbol, bv=bv, parent_addr=symbol.address)
-                return UnaryOperation(OperationType.address, [lifted_global_var])
-            else:
-                # otherwise it is an IAT entry -> construct ImportedFunctionSymbol with the same name.
-                return ImportedFunctionSymbol(symbol.name, symbol.address, vartype=Pointer(Integer.char()))
+    def lift_constant_pointer(self, pointer: mediumlevelil.MediumLevelILConstPtr, **kwargs) -> Constant:
+        """Lift the given constant pointer, e.g. &0x80000.
+            For clarity all cases:
+                1. NULL (&0x0)
+                    - is the definition of NULL in C, therefore a separate case (otherwise elf/dos header lifting)
+                2. Address has datavariable with a basic type (everything except void/void*)
+                    - lift as datavariable
+                3. Address has a symbol, which is not a datasymbol
+                    - lift as symbol
+                4. Address has a function there 
+                    - lift the function symbol as symbol
```

Comment:
We can do that. 
I mean the "last case", everything else, will always results in a void/char pointer. 

But what would be a "good" condition to check for?

Reply:
That's the problem - we could use a sorted directory and have a `lambda x: True` condition for the case at the end, but it kinda does not feel good. Just want to avoid feature creep in this function, so extracting the first four cases into their own function will make do for now.

Reply:
I split the function without a new handler case. 
The new function is just the last part of the former one, which lifts a given "raw address".